libs/proposals/src/lib/proposals-hooks/use-market-view-proposals.ts

```diff
@@ -0,0 +1,29 @@
+import { useDataProvider } from '@vegaprotocol/data-provider';
+import { marketViewProposalsDataProvider } from '../proposals-data-provider';
+import type * as Types from '@vegaprotocol/types';
+
+export const useMarketViewProposals = ({
+  skip = false,
+  typename,
+  proposalType,
+  inState,
+}: {
+  typename: string;
```

Comment:
Derive a type from the schema...

```suggestion
  typename: Types.ProposalChange['__typename']
```